lib/Email/Stuffer/TestLinks.pm

```diff
@@ -0,0 +1,77 @@
+package Email::Stuffer::TestLinks;
+
+use strict;
+use warnings;
+
+our $VERSION = '0.010';
+
+use Test::Most;
+use Mojo::UserAgent;
+use Email::Stuffer;
+use Class::Method::Modifiers qw/ install_modifier /;
+
+=head1 SYNOPSIS
+
+    use Email::Stuffer::TestLinks;
+
+=head1 NAME
+
+Email::Stuffer::TestLinks - validates links in HTML emails sent by Email::Stuffer>send_or_die()
+
+=head1 DESCRIPTION
+
+When this module is included in a test, it parses HTML links (<a href="xyz"...) in every email sent through Email::Stuffer->send_or_die(). Each URI must get a successful response code (200 range) and the returned pagetitle must not contain 'error' or 'not found'.
+
+=cut
+
+install_modifier 'Email::Stuffer', after => send_or_die => sub {
+
+    my $self = shift;
+    my $ua   = Mojo::UserAgent->new;
+    $ua->max_redirects(10);
+    $ua->connect_timeout(5);
+
+    my %urls;
+    my $body;
+    $self->email->walk_parts(
+        sub {
+            my ($part) = @_;
+            if ($part->content_type && $part->content_type =~ /text\/html/i) {
```

Comment:
You can early `return unless $part->content_type =~ m[text/html]i;` and proceed with the link finding, instead of an `if` block.

Reply:
Thanks, done